content/blog/2022-07-05-Practical-data-drift-monitoring-with-DVC.md

```diff
@@ -0,0 +1,337 @@
+  this blog post and set up an automatic drift detector as part of the ML
+  pipeline. Quite a few libraries, even open-source, provide  users with drift
+  detection capabilities. We will show how we may integrate  them into the ML
+  system with DVC and how, in general, DVC features are  beneficial for data
+  drift detection.
+picture: TODO
+author: dan_martinec
+commentsUrl: TODO
+
+tags:
+  - Data Validation
+  - Machine Learning
+  - MLOps
+  - Productionization
+---
+
+# Introduction
+
+## Data distribution shift
+
+There is a large boom of ML systems deployed to production these days. This
+brings many demands to ML engineers. ML pipeline automation is possibly the most
+important one. However, there is also one less known but very important aspect.
+That is the validation of inputs and outputs of the ML system. In fact, data
+validation is listed as one of the
+[hidden technical debts in machine learning systems](https://proceedings.neurips.cc/paper/2015/file/86df7dcfd896fcaf2674f757a2463eba-Paper.pdf).
+There is a long history of projects that end up in a disaster and that could be
+most likely saved by a proper data validation process.
+
+One of the common problems is a data distribution shift. In short, the data the
+model uses has significantly changed over time. This results in poor performance
+of an ML model. Therefore, detecting a data distribution shift is essential as
+soon as possible. There are several types of data distribution shifts. We will
+discuss two: a) covariate shift which we’ll cover later and b) concept drift,
+which you can see in the figure below. The figure shows the average number of
+rented bikes during non-working hours for the last two weekends of April and the
+first weekend in May (blue line) vs. the second weekend in May (orange line). We
+can see a significant change of pattern in the afternoon hours.
+
+![Example of concept drift](/uploads/images/2022-07-05/example_of_concept_drift.png '=800')
+
+Data validation is a vast topic to cover. We refer to an excellent
+[Data Distribution Shifts and Monitoring post](https://huyenchip.com/2022/02/07/data-distribution-shifts-and-monitoring.html)
+for a detailed overview and precise definition of covariate shift and concept
+drift.
+
+## Project description
+
+We demonstrate the data validation concept on a
+[bike-sharing dataset](https://archive.ics.uci.edu/ml/datasets/bike+sharing+dataset)
+(same as in
+[Preventing Stale Models in Production](https://dvc.org/blog/stale-models)) from
+a bike-rental company. The tricky part about this project is that it is highly
+correlated to the environmental and seasonal settings. This makes both the ML
+model and the drift detector training rather complicated as the input data is
+changing all the time, and we need to consider it during ML system design.
+
+| Timestamp        | temp | hum  | windspeed | hour | weekday | season | holiday | workingday | cnt |
```

Comment:
Can we wrap this in a code block (i.e. using something like ```dvctable```? 
It renders kind of strange as it is because it excels in the margins.